cmeutils/structure.py

```diff
@@ -50,6 +49,88 @@ def get_quaternions(n_views = 20):
     unit_z = np.array([0, 0, 1])
     return [vector_vector_rotation(i, unit_z) for i in points]
 
+def all_atom_rdf(gsdfile,
+                 start=0,
+                 stop=None,
+                 r_max=None,
+                 r_min=0,
+                 bins=100,
+                 ):
+    """ Parameters
+    ----------
+    gsdfile : str
+        Filename of the GSD trajectory.
+    A_name, B_name : str
+        Name(s) of particles between which to calculate the RDF (found in
+        gsd.hoomd.Snapshot.particles.types)
+    start : int
+        Starting frame index for accumulating the RDF. Negative numbers index
+        from the end. (default 0)
+    stop : int
+        Final frame index for accumulating the RDF. If None, the last frame
+        will be used. (default -1)
+    r_max : float
+        Maximum radius of RDF. If None, half of the maximum box size is used.
+        (default -1)
+    r_min : float
+        Minimum radius of RDF. (default 0)
+    bins : int
+        Number of bins to use when calculating the RDF. (default 100)
+
+    Returns
+    -------
+    (freud.density.RDF, float)
+    """
+
+    if not stop:
+        stop = -1
+
+    with gsd.hoomd.open(gsdfile, mode="rb") as trajectory:
+        snap = trajectory[0]
+
+        if r_max is None:
+           #Use a value just less than half the maximum box length.
+            r_max = np.nextafter(
+            np.max(snap.configuration.box[:3]) * 0.5, 0, dtype=np.float32
+            )
```

Comment:
Because we usually run our simulations starting with a shrink step, if the first frame is before shrink and the last frame is after shrink, this logic could cause us to run into trouble... (I know this already existed in the other RDF function, but maybe it shouldn't be that way.) I lean towards not having a default for rmax.